dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/WebSphereCDIDeploymentImpl.java

```diff
@@ -565,38 +566,83 @@ public WebSphereBeanDeploymentArchive getBeanDeploymentArchive(Class<?> beanClas
 
     @Override
     @Trivial
-    //This method is called a lot
     public WebSphereBeanDeploymentArchive getBeanDeploymentArchiveFromClass(Class<?> clazz) {
-        // Note this method looks for a BDA which has the given class in it, **whether the class is a bean or not**
-        // We needed this so that we can find the correct bean manager for non-beans like servlets
+
         WebSphereBeanDeploymentArchive wbda = classBDAMap.get(clazz);
 
         if (wbda == null) {
-            for (WebSphereBeanDeploymentArchive bda : orderedBDAs) {
-                if (bda.getAllClazzes().contains(clazz.getName())) {
+            wbda = orderedBDAs.stream()
+                              .filter(bda -> doesBDAContainClazz(bda, clazz))
+                              .findFirst()
+                              .get();
+
+            if (wbda != null) {
+                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                    Tr.debug(tc, "mapping class: " + clazz.getName() + " " + System.identityHashCode(clazz) + " to bda: " + wbda.toString());
+                }
```

Comment:
`.get()` here will throw an exception if none was found.

Since this method is called a lot, it might be worth keeping this as a `for` loop since streams tend to be slightly slower.